### PR TITLE
Feature/dropwizard out of framework

### PR DESCRIPTION
**Pull request by tcrossrfid**

### CODE REVIEW

File: smartcosmos-extension-starter-parent/smartcosmos-extension-archetype/src/main/resources/archetype-resources/pom.xml [hunk 1]
```diff
@@ -31,8 +31,16 @@
             <artifactId>smartcosmos-extension-starter-test</artifactId>
             <scope>test</scope>
         </dependency>
+
+        <dependency>
+            <groupId>net.smartcosmos</groupId>
+            <artifactId>smartcosmos-objects-developer-edition</artifactId>
```
Comment (voor) [inline smartcosmos-extension-starter-parent/smartcosmos-extension-archetype/src/main/resources/archetype-resources/pom.xml]: Can't do this.

File: smartcosmos-extension-api/src/main/java/net/smartcosmos/platform/api/dao/IAbstractNamedObjectDAO.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import java.util.Collection;
+
+/**
+ * Initially created by tcross on November 16, 2015.
+ */
+public interface IAbstractNamedObjectDAO <U extends INamedObject<U>, V extends U> extends IAbstractDAO<U, V>
+
+{
+
+    Collection<U> findByNameLike(Class<?> clazz, String nameLike, IAccount account);
```
Comment (voor) [inline smartcosmos-extension-api/src/main/java/net/smartcosmos/platform/api/dao/IAbstractNamedObjectDAO.java]: They won't have access to the Class, so this needs to not be in the interface, and then appropriately utilize getEntityClass() in the implementation.  That needs to be changed everywhere else it's shown.

File: smartcosmos-extension-api/src/main/java/net/smartcosmos/platform/api/dao/IBaseDAO.java [hunk 3]
```diff
@@ -33,13 +33,13 @@
  */
 public interface IBaseDAO<T extends IDomainResource<T>> extends IMonikerSearchDAO<T>, IAdvancedQuery<T>
 {
-    T findByUrn(Class<?> clazz, String urn);
+    T findByUrn(Class clazz, String urn);
```
Comment (voor) [inline smartcosmos-extension-api/src/main/java/net/smartcosmos/platform/api/dao/IBaseDAO.java]: See earlier comment -- they won't have this class, so they can't pass it in.

**Comment by tcrossrfid** [comment]
Cleaned up everywhere (although not yet committed/pushed) except for one spot: in the archetype, in ExampleExtensionEndpointsResourceRegistrar, I need to call
context.getEnvironment().jersey().register(new ExampleExtensionResource(context));
to register the example resources, and starting at getEnvironment(), it's all Dropwizard. I'm not seeing a way to separate the chicken from the egg, and still have an archetype that will build right out of the box. I'll keep staring at it, but if you have a suggestion, I'd love to hear it.

**Comment by voor** [comment]
Going to re-evaluate this and put it on the roadmap for 2016.  This pull request is going to continue getting out of date.

**Pull request closed without merging**